handlers/stock.js

```diff
@@ -0,0 +1,176 @@
+        });
+    }
+
+    // Show the stock for all users
+    if (splitParams[0] === 'allbalances') {
+        showAllStock(tableSvc, channel);
+    }
+
+    // Send stock to another user if they have an account
+    if (splitParams[0] === 'send') {
+        const recipient = splitParams[1];
+        const sendAmount = parseInt(splitParams[2]);
+
+        if (recipient !== message.author.username) {
+            getStock(tableSvc, recipient, function (receiverStock) {
+                if (typeof receiverStock === 'number') {
+                    getStock(tableSvc, message.author.username, function (senderStock) {
+                        if ((senderStock - sendAmount) > 0 && typeof senderStock === 'number') {
+
+                            const newSenderTotal = senderStock - sendAmount;
+                            const newReceiverTotal = receiverStock + sendAmount;
+                            let transactionString = 'Transaction Complete:' + '\n';
+
+                            updateUser(tableSvc, message.author.username, newSenderTotal);
+                            updateUser(tableSvc, recipient, newReceiverTotal);
+
+                            transactionString += `${message.author.username}: ${senderStock} - ${sendAmount} = ${newSenderTotal}\n
+                                                  ${recipient}: ${receiverStock} + ${sendAmount} = ${newReceiverTotal}`
+
+                            LOG.info(transactionString);
+                            channel.send(transactionString);
+                        }
+                        else {
+                            channel.send('Not enough stock to make the transaction.')
+                        }
+                    })
+                }
+                else {
+                    channel.send('That user does not exist! Schreef won\'t throw his stock to the abyss.');
+                }
+            })
+        }
+        else {
+            channel.send('You can\'t send yourself stock here.');
+        }
+    }
+
+};
+
+// Used to update or create a user entry in the Azure Table
+function updateUser(tableSvc, username, amount) {
+
+    let userEntry = {
+        PartitionKey: ENTGEN.String(username),
+        RowKey: ENTGEN.String('1'),
+        schreefstock: ENTGEN.Int32(amount),
+    }
+
+    tableSvc.insertOrReplaceEntity(TABLENAME, userEntry, null, function (error, result, response) {
+        if (!error) {
+            // Entity updated
+        }
+        else {
+            LOG.error(error);
+        }
+    })
+};
+
+// Displays everyone's stock
+function showAllStock(tableSvc, channel) {
+    const query = new azure.TableQuery().where('RowKey eq ?', '1');
+    let displayResult = 'Full ledger: ';
+
+    tableSvc.queryEntities(TABLENAME, query, null, function (error, result, response) {
+        if (!error) {
+            // query successful
+            for (let i = 0, len = result.entries.length; i < len; i++) {
```

Comment:
can you not use forEach or map on result.entries.length?

Reply:
probably can - i'll rewrite it later today